docs/kusion/3-concepts/6-spec.md

```diff
@@ -20,3 +20,101 @@ Delivering an application to different environments with identical configuration
+        status: {}
+      extensions:
+        GVK: /v1, Kind=Namespace
+    - id: apps/v1:Deployment:quickstart:quickstart-default-quickstart
+      type: Kubernetes
+      attributes:
+        apiVersion: apps/v1
+        kind: Deployment
+        metadata:
+            creationTimestamp: null
+            labels:
+                app.kubernetes.io/name: quickstart
+                app.kubernetes.io/part-of: quickstart
+            name: quickstart-default-quickstart
+            namespace: quickstart
+        spec:
+            selector:
+                matchLabels:
+                    app.kubernetes.io/name: quickstart
+                    app.kubernetes.io/part-of: quickstart
+            strategy: {}
+            template:
+                metadata:
+                    creationTimestamp: null
+                    labels:
+                        app.kubernetes.io/name: quickstart
+                        app.kubernetes.io/part-of: quickstart
+                spec:
+                    containers:
+                        - image: kusionstack/kusion-quickstart:latest
+                          name: quickstart
+                          resources: {}
+        status: {}
+      dependsOn:
+        - v1:Namespace:quickstart
+        - v1:Service:quickstart:quickstart-default-quickstart-private
+      extensions:
+        GVK: apps/v1, Kind=Deployment
+    - id: v1:Service:quickstart:quickstart-default-quickstart-private
+      type: Kubernetes
+      attributes:
+        apiVersion: v1
+        kind: Service
+        metadata:
+            creationTimestamp: null
+            labels:
+                app.kubernetes.io/name: quickstart
+                app.kubernetes.io/part-of: quickstart
+            name: quickstart-default-quickstart-private
+            namespace: quickstart
+        spec:
+            ports:
+                - name: quickstart-default-quickstart-private-8080-tcp
+                  port: 8080
+                  protocol: TCP
+                  targetPort: 8080
+            selector:
+                app.kubernetes.io/name: quickstart
+                app.kubernetes.io/part-of: quickstart
+            type: ClusterIP
+        status:
+            loadBalancer: {}
+      dependsOn:
+        - v1:Namespace:quickstart
+      extensions:
+        GVK: /v1, Kind=Service
+secretStore: null
+context: {}
+```
+
+From the example above, we can see that the `Spec` contains a list of `resources` required by the application. Each resource in the `Spec` needs to have `id`, `type`, `attributes`, `dependsOn`, and `extensions` fields: 
+
+- `id` is the unique key of this resource. An idiomatic way for Kubernetes resources is `apiVersion:kind:namespace:name`, and for Terraform resources is `providerNamespace:providerName:resourceType:resourceName`. 
+- `type` represents the type of runtime Kusion supports, and currently includes `Kubernetes` and `Terraform`. 
+- `dependsOn` contains all the other resources the resource depends on. 
```

Comment:
please add an explanation about the field `attribute`

Reply:
OK, I've added a brief explanation about the field `attributes` in the doc.